framework/src/org/checkerframework/framework/util/AnnotationMirrorMap.java

```diff
@@ -0,0 +1,111 @@
+package org.checkerframework.framework.util;
+
+import static org.checkerframework.javacutil.AnnotationUtils.annotationOrdering;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import javax.lang.model.element.AnnotationMirror;
+import org.checkerframework.javacutil.AnnotationUtils;
+
+/**
+ * Implements the Map interface where the keys to the map are AnnotationMirrors. This classes uses
+ * {@link AnnotationUtils#areSame(AnnotationMirror, AnnotationMirror)} instead of {@link
+ * AnnotationMirror#equals(Object)}.
```

Comment:
Regarding "instead of", please explicitly say where this class came from -- in particular, did you copy some existing implementation (in which case give credit and also put a comment at the occurrences of areSame), or is it a clean-room reimplementation?